File: others/WorksApplication/Worksap Package/workspace/China_Intern/China_Intern_Java/src/jp/co/worksap/intern/entities/staff/StaffDTOReader.java
package jp.co.worksap.intern.entities.staff;

import java.io.IOException;

import jp.co.worksap.intern.constants.Constants;
import jp.co.worksap.intern.entities.AbstractDTOReader;

public class StaffDTOReader extends AbstractDTOReader<StaffDTO> {
	private static final int COLUMN_INDEX_STAFF_ID = 0;
	private static final int COLUMN_INDEX_NAME = 1;
	private static final int COLUMN_INDEX_GENDER = 2;
	private static final int COLUMN_INDEX_RANK = 3;
	private static final int COLUMN_INDEX_POSITION = 4;
	private static final int COLUMN_INDEX_SUPERMARKET_ID = 5;
	

	private String fileAddress = Constants.DEFAULT_CSV_FOLDER+"STAFF_MST.csv";

	/**
	 * use default file address
	 * 
	 * @throws IOException
	 */
	public StaffDTOReader() throws IOException {
		super(StaffDTOReader.class.getName());
		super.init();
	}

	/**
	 * use customize file address
	 * 
	 * @param fileAddress
	 * @throws IOException
	 */
	public StaffDTOReader(final String fileAddress) throws IOException {
		super(StaffDTOReader.class.getName());
		this.fileAddress = fileAddress;
		super.init();
	}

	@Override
	protected String getFileArress() {
		return fileAddress;
	}

	@Override
	protected StaffDTO convertArrayToDTO(String[] value) throws IOException {

		Long staffId = Long.valueOf(value[COLUMN_INDEX_STAFF_ID]);
		String name = value[COLUMN_INDEX_NAME];
		String gender = value[COLUMN_INDEX_GENDER];
		Long supermarketId = Long.valueOf(value[COLUMN_INDEX_SUPERMARKET_ID]);
		RankType rank = RankType.valueOfString(value[COLUMN_INDEX_RANK]);
		PositionType position = PositionType.valueOfString(value[COLUMN_INDEX_POSITION]);
		
		StaffDTO dto = new StaffDTO(staffId, name, gender, rank, position, supermarketId);
		return dto;
	}
}

File: others/WorksApplication/Worksap Package/workspace/China_Intern/China_Intern_Cpp/src/entities/staff/staff_dto_reader.cpp
#include "staff_dto_reader.h"
#include "staff_dto.h"
#include "../../constants/constants.h"
#include "../icsv_master_dto.h"
#include "rank_type.h"
#include "position_type.h"
#include <string>
#include <vector>
#include <cstdlib>
using namespace std;

StaffDTOReader::StaffDTOReader() {
	this->file_address = kDefaultCsvFolder + "STAFF_MST.csv";
	AbstractDTOReader::init();
}

StaffDTOReader::StaffDTOReader(string &file_address) {
	this->file_address = file_address;
	AbstractDTOReader::init();
}

string StaffDTOReader::get_file_address() {
	return this->file_address;
}

ICsvMasterDTO *StaffDTOReader::convert_array_to_dto(vector<string> &value) {
	long staff_id = atol(value[COLUMN_INDEX_STAFF_ID].c_str());
	string name = value[COLUMN_INDEX_NAME];
	string gender = value[COLUMN_INDEX_GENDER];
	RankType rank = rank_type_value_of_string(value[COLUMN_INDEX_RANK]);
	PositionType position = position_type_value_of_string(
			value[COLUMN_INDEX_POSITION]);
	long supermarket_id = atol(value[COLUMN_INDEX_SUPERMARKET_ID].c_str());
	ICsvMasterDTO *obj = new StaffDTO(staff_id, name, gender, rank,
			position, supermarket_id);
	return obj;
}

StaffDTOReader::~StaffDTOReader() {

}

